libflux/fluxdoc/SPEC.md

```diff
@@ -0,0 +1,284 @@
+
+Because this value will only ever be treated as a string in the docs, writing a description of the constant's value, rather than the literal value, is acceptable if it makes things easier to read or understand.
+
+If a constant is not defined with a builtin statement, the doc comment parser should be able to get its value from the flux code itself, and no documentation is required.
+
+_Example:_
+
+```
+// 0.693147180559945309417232121458176568075500134360255254120680009
+builtin ln2 : float
+
+// 1 ÷ math.ln2
+builtin log2e : float
+
+// 2.30258509299404568401799145468436420760110148862877297603332790
+builtin ln10 : float
+
+// 1 ÷ math.ln10
+builtin log10e : float
+
+// 1.797693134862315708145274237317043567981e+308
+builtin maxfloat : float
+
+// 4.940656458412465441765687928682213723651e-324
+builtin smallestNonzeroFloat : float
+
+// 1<<63 - 1
+builtin maxint : int
+
+// -1 << 63
+builtin minint : int
+
+// 1<<64 - 1
+builtin maxuint : uint
+```
+
+## Documenting Groups of Constants
+
+Some packages contain multiple constants that are meaningfully related to one another. In these cases, it may be helpful to document them as a group, rather than document each one of them separately.
+
+A doc comment for a group of constants consists of a heading and a description. The heading should be a markdown H1 heading, and should be immediately followed by a blank line comment. The description should be a single line of text explaining what the constants represent and/or how they are related.
+
+There should be no blank lines or lines containing code between the declaration/definition of one constant in a group and the declaration of the next constant in the same group.
+
+If one of the constants in a group is defined using a `builtin` statement, a doc comment should be added on the line immediately above its declaration (See the `Documenting Builtin Constants` section above).
+
+The end of a constant group is deliniated by a blank, uncommented line.
+
+_Example:_
+
+```
+// # Days of the week
+//
+// The days of the week are represented as integers in the range `[0-6]`
+Sunday = 0
+Monday = 1
+Tuesday = 2
+Wednesday = 3
+Thursday = 4
+Friday = 5
+Saturday = 6
+
+// # Months of the year
+//
+// Months are represented as integers in the range `[1-12]`
+January = 1
+February = 2
+March = 3
+April = 4
+May = 5
+June = 6
+July = 7
+August = 8
+September = 9
+October = 10
+November = 11
+December = 12
+```
+
+## Docs as Rust Data Structures
+
+The overarching goal for this project is to be able to parse the new docs into rust structures, which we will then serialize into JSON for broader consumption.
+
+Here's a proprosal for what those rust data structures could look like:
+
+```rust
+enum Doc {
```

Comment:
I just had another thought: what if we add source location information to all of these structs? Then we could link back to the github source code from the docs. 

This doesn't need to be in the first implementation but something we could add later.

Reply:
That would be great. It might be the small push someone needs to start contributing code!